### PR TITLE
fix(button): add icon/text wrapper

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly/issues/5944
fixes https://github.com/patternfly/patternfly/issues/6245

~~[BackstopJS Report-6.18.24.pdf](https://github.com/user-attachments/files/15891164/BackstopJS.Report-6.18.24.pdf)~~

Backstop visual regression report - [backstop-new-new-6.18.24.pdf](https://github.com/user-attachments/files/15892839/backstop-new-new-6.18.24.pdf)

[First commit](https://github.com/patternfly/patternfly/pull/6806/commits/ff4c7e2822c0298283b10dc4333a9b1788050338) is the CSS change if you want to look at just that. [Second commit](https://github.com/patternfly/patternfly/pull/6806/commits/1ee915881aa07f924420c4040ea39cbeea83641e) is the hbs updates.

I only updated the button component examples/demos (not in any of the other  but the backstop report looks good. A follow up would be to update the use of button in other component examples/demos.

TL;DR of the hbs change creates a similar API to the proposed react component change. The `{{#> button}}`'s `@partial-block` (or "children" in react) will always be the button text. Any time an icon is used, we'll pass it as an hbs attr/prop. With the change in hbs, that means all the other stuff (like count or whatever) will also need to be added via an hbs attr/prop.

### CODE REVIEW

**Review by srambach — APPROVED** [review]
OK, looks good with the little inline fix! 😹

**Review by thatblindgeye — APPROVED** [review]

**Review by andrew-ronaldson — APPROVED** [review]
it took my breath away.

**Review by srambach — APPROVED** [review]
Pushes all the right buttons
